### ISSUE TITLE
[bug] wireguard with cillium example doesn't work 

### ISSUE THREAD

**Issue by csantanapr** [aws-ia/terraform-aws-eks-blueprints#1618]
## Description

When using the example [wireguard-with-cilium](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/refactor/addons-module/examples/wireguard-with-cilium) the example pod `client` is not communicating because is configured before cilium wireguard is established.

we need to put some type of sleep/delay before creating the client pod

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

git clone branch
cd examples/wireguard-with-cilium/
terraform init
terraform apply -auto-approve

follow readme and run `tcpdump -A -c 3 -i cilium_wg0` command

## Expected behaviour

For `tcpdump -A -c 3 -i cilium_wg0` to show encrypted traffic

## Actual behaviour

The `tcpdump -A -c 3 -i cilium_wg0`coammand doesn't show any output

## Additional context

Thank you to @aaroniscode and the blog [Transparent encryption of node to node traffic on Amazon EKS using WireGuard and Cilium](https://aws.amazon.com/blogs/containers/transparent-encryption-of-node-to-node-traffic-on-amazon-eks-using-wireguard-and-cilium/) he authored to debug the issue

The steps on the blog uses eksctl and there is more time between installing cilium and deploying the client pod, this is why it works if you use the blog instructions

**Comment by csantanapr** [comment]
It looks like is not a race condition, is about not allowing UDP/51871 communication between nodes, cilium wireguard is configured to tunnel all encryption data thru this port

I tested the example disabling the creation of security groups in eks module and it worked out of the box
```hcl
  create_node_security_group = false
```

Now testing adding the udp/51871 ingress, because I don't we want to recommend to open everything up between the nodes
```hcl
  # Extend node-to-node security group rules
  node_security_group_additional_rules = {

    ingress_cilium_wireguard = {
      description = "Allow Cilium Wireguard node to node"
      protocol    = "udp"
      from_port   = 51871
      to_port     = 51871  # Cilium Wireguard Port https://github.com/cilium/cilium/blob/main/Documentation/security/network/encryption-wireguard.rst
      type        = "ingress"
      self        = true
    }
  }
```
cc @askulkarni2 @aaroniscode

**Comment by csantanapr** [comment]
Allowing traffic is not enough to get encrypted traffic working.

I tried with with all default security groups and additional one for `udp/51871` with the client/server pods deployed with terraform together with cilium and the traffic was flowing but **NOT** thru the wrireguard tunnel.

Then I tried with with all default security groups and additional one for `udp/51871` **without** the client/server pods deployed with terraform together with cilium. Then waited ~5mins then deploy the client/server pods and then traffic was flowing thru wireguard tunnel.
